.github/workflows/tests.yaml

```diff
@@ -9,48 +9,49 @@ on:
     branches:
       - main
 
+# ensuring only one instance is running at a given time
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   test-conda-store-server:
     name: "unit-test conda-store-server"
     strategy:
       matrix:
         # cannot run on windows due to needing fake-chroot for conda-docker
-        os: [ubuntu-latest, macos-latest]
-    runs-on: ${{ matrix.os }}
+        os: ["ubuntu", "macos"]
+        include:
+          - os: ubuntu
+            environment-file: conda-store-server/environment-dev.yaml
+            miniforge-version: latest
+          - os: macos
+            environment-file: conda-store-server/environment-macos-dev.yaml
+            miniforge-version: latest
```

Comment:
Again this _should_ not matter much. We are pinning the dev and runtime dependencies, which are the ones that really matter. 
`miniforge` do frequent releases, and from experience, versions in files like this (e.g., actions versions) are easily forgotten by people and rarely updated. Since this is not a critical dependency nor a user-facing one I'd prefer not pinning - we can very easily see the version from the output logs if needed for debugging.